web/client/utils/LayersUtils.js

```diff
@@ -125,6 +125,52 @@ const getGroupNodes = (node) => {
 };
 
 
+const sortNestedGroups = (groups, configGroups) => {
```

Comment:
please add jsdoc for this function and also we need unit test for this.
i suggest to write the test using tdd
- so you simply comment out your changes
- write the test that will fail
- reapply your changes
- verify now the test is successful